src/NServiceBus.Core/Endpoint.cs

```diff
@@ -1,21 +1,47 @@
 namespace NServiceBus
 {
     using System.Threading.Tasks;
+    using ObjectBuilder;
 
     /// <summary>
     /// Provides factory methods for creating and starting endpoint instances.
     /// </summary>
     public static class Endpoint
     {
+        internal static PreparedEndpoint Prepare(EndpointConfiguration configuration, IConfigureComponents configureComponents)
+        {
+            Guard.AgainstNull(nameof(configuration), configuration);
+            Guard.AgainstNull(nameof(configureComponents), configureComponents);
+
+            configuration.UseExternallyManagedContainer(configureComponents);
+
+            var initializable = configuration.Build();
+
+            return initializable.Prepare();
+        }
+
+        internal static async Task<IEndpointInstance> Start(PreparedEndpoint preparedEndpoint, IBuilder builder)
```

Comment:
@danielmarbach this is where you want us to add a cancellation token right?

Reply:
yes and then move it down the other await statements if possible. Ideally if we are touching those APIs we could also introduce a public api change that is minor bump compatible. For example

```
public static async Task<IStartableEndpoint> Create(EndpointConfiguration configuration, CancellationToken cancellationToken = default)
```

to have it on the public APIs ready